internal/status/gql_reporter.go

```diff
@@ -36,15 +36,42 @@ type GraphQLStatusReporter struct {
 	resVerMutex     sync.RWMutex
 }
 
-func newGraphQLStatusReporter(logger logrus.FieldLogger, client GraphQLClient, resVerClient ResVerClient, cfgVersion int) *GraphQLStatusReporter {
+func newGraphQLStatusReporter(logger logrus.FieldLogger, client GraphQLClient, resVerClient ResVerClient) *GraphQLStatusReporter {
 	return &GraphQLStatusReporter{
-		log:             logger,
-		gql:             client,
-		resVerClient:    resVerClient,
-		resourceVersion: cfgVersion,
+		log:          logger,
+		gql:          client,
+		resVerClient: resVerClient,
 	}
 }
 
+// ReportDeploymentConnectionInit reports connection initialization.
+func (r *GraphQLStatusReporter) ReportDeploymentConnectionInit(ctx context.Context, k8sVer string) error {
+	logger := r.log.WithFields(logrus.Fields{
+		"deploymentID": r.gql.DeploymentID(),
+		"type":         "connecting",
+	})
+	logger.Debug("Reporting...")
+
+	err := r.withRetry(ctx, logger, func() error {
+		var mutation struct {
+			Success bool `graphql:"reportDeploymentConnectionInit(id: $id, botkubeVersion: $botkubeVersion, k8sVer: $k8sVer)"`
+		}
+		variables := map[string]interface{}{
+			"id":             graphql.ID(r.gql.DeploymentID()),
+			"botkubeVersion": version.Info().Version,
+			"k8sVer":         k8sVer,
+		}
```

Comment:
**nit:** this is static, so it can be extracted from the retry func.

Reply:
TBH I don't see relation between static and retry. And what do you mean that "this is static"?

Reply:
By static, I mean that they don't change between each retry execution. By having it in the retry function, you are going to always recreate it and GC will need to clean the previously allocated variables.

Because of that, you can just have it outside the retry and use closure function. 

However, it not that big deal, so that's the reason I used also "**nit**" 🙂

Reply:
Make sens. Fixed